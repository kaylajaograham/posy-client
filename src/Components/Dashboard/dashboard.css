.dash_hero {
    margin: 0 auto;
    text-align: center;
    position: relative;
    width: 100vw;
    height: 75vh;
    
    /* Flexbox stuff */
    display: flex;
    justify-content: center;
    align-items: center;
    
    /* Text styles */
    text-align: center;
    background-image: url(https://images.unsplash.com/photo-1559825481-12a05cc00344?ixlib=rb-1.2.1&auto=format&fit=crop&w=700&q=60);
    background-size: cover;
    background-position: center center;
    background-repeat: no-repeat;
    background-attachment: fixed;
}

h3 {
    margin: 0px;
    background: none;
    color: white;
    display: flexbox;
    text-shadow: 2px 2px 3px #0B1C48;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 50%;
    height: 30%;
    margin: auto;
}

h4 {
    font-family: tahoma;
    font-size: 32px;
    color: #0B1C48;
    text-shadow: 1px 1px 3px gray;

}

img {
    height: auto;
    width: 100%;
}

p {
    padding-bottom: 25px;
}

/* INDIVIDUAL LOG LAYOUT */
.addLogButton {
    border-radius: 15px;
    padding: 8px 15px;
    text-decoration: none;
    font-weight: bold;
    color: white;
    border-right: 1px solid #ccc; 
    margin: 50px;
    background-color: peru;
}

.button {
    font-size: 14px;
    
    
}

