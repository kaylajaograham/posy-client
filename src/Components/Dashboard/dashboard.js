import React, {Component} from 'react';
import './dashboard.css';
import { Link } from 'react-router-dom';
import Nav from '../Nav/nav';
import PosyContext from '../../PosyContext';
import Log from '../Log/Log';
import {findLog} from '../../log-helpers';


export default class Dashboard extends React.Component {
    state = {
      forErrors: this.props.match,
      toggle: true
    }
  
    static defaultProps = {
      match: {
        params: {}
      }
    }
  
    static contextType = PosyContext;
  
    // sends user back to homepage after delete
    handleDeleteLog = logId => {
      this.props.history.push('/')
    }

    render() {
        const { logs=[] } = this.context
        const { logId } = this.state.forErrors.params
        const log = findLog(logs, logId) || { content: ''}
          if(this.state.toggle === false) {
            this.setState({
              forErrors: 'err'
            })
            this.setState({
              forErrors: this.props.match
            })
          }

        return (
            <div className='dashboard'>
                <Nav />
                <div className='dash_hero'>
                        <h3>"Be mindful; be thinkful"</h3>
                </div>

                <div className='dash_main'>
                        <h4>Welcome back, Kayla.</h4>
                        <p>Take a few minutes to jot down something on your mind. Tag your entry to track any patterns.</p>
                    
                    <div className='dash_logs'>
                    
    
                        <Link className='addLogButton' to='/addlog'>New Entry</Link>
                
                        <section className='LogList'>
                            <Log
                            id={log.id}
                            log_name={log.log_name}
                            log_tag={log.log_tag}
                            log_entry={log.log_entry}
                            onDeleteLog={this.handleDeleteLog}
                            />
                        </section>

                    </div>
                </div>
            </div>
        )
    }
} 


/*
                        <div className='dash_log_box'>
                                <h5>I need a vacation</h5>
                                 <h6>Inspired</h6>
                        </div>

                        <div className='dash_log_box'>
                                <h5>I didn't get the job</h5>
                                 <h6>Positivity</h6>
                        </div>

                        <div className='dash_log_box'>
                                <h5>Anniversary</h5>
                                 <h6>Loved</h6>
                        </div>

                        <div className='dash_log_box'>
                                <h5>Cafe chat with a stranger</h5>
                                 <h6>Inspired</h6>
                        </div>

                        <div className='dash_log_box'>
                                <h5>Goals</h5>
                                 <h6>Motivated</h6>
                        </div>

                        */