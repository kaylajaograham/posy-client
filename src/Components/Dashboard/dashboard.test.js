import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dashboard from './dashboard';

describe(`dashboard component`, () => {
  const props = {
    itinerary: {
      "id": 1,
      "log": "test"
    }
  };

  it('renders the ItineraryList given props', () => {
    const wrapper = shallow(<Dashboard {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})