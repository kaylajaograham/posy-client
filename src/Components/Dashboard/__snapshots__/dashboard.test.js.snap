// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard component renders the ItineraryList given props 1`] = `
<div
  className="dashboard"
>
  <div
    className="dash_hero"
  >
    <h3>
      "Be mindful; be thinkful"
    </h3>
  </div>
  <div
    className="dash_main"
  >
    <h4
      className="welcome"
    >
      N A M A S T E
    </h4>
    <p>
      Take a few minutes to jot down something on your mind. Rid yourself of negative energy to make space for positivity.
    </p>
    <div
      className="dash_logs"
    >
      <Link
        className="addLogButton"
        to="/addlog"
      >
        New Entry
      </Link>
      <hr />
      <section
        className="LogList"
      >
        <h4
          className="yourLogs"
        >
          Your logs
        </h4>
        <LogList
          match={
            Object {
              "params": Object {},
            }
          }
        />
      </section>
    </div>
  </div>
</div>
`;
